openchs-server-api/src/main/java/org/openchs/web/EncounterTypeController.java

```diff
@@ -182,12 +182,19 @@ private FormMapping createFormMapping() {
     @Transactional
     public ResponseEntity voidEncounterType(@PathVariable("id") Long id) {
         OperationalEncounterType operationalEncounterType = operationalEncounterTypeRepository.findOne(id);
+        List<FormMapping> formMappings = formMappingRepository.findAllByIsVoidedFalse();      
         if (operationalEncounterType == null)
             return ResponseEntity.notFound().build();
         EncounterType encounterType = operationalEncounterType.getEncounterType();
         if (encounterType == null)
             return ResponseEntity.notFound().build();
 
+        for (int formMapping = 0; formMapping < formMappings.size(); formMapping++) {
+            if(formMappings.get(formMapping).getEncounterTypeUuid() != null && formMappings.get(formMapping).getEncounterTypeUuid().equals(encounterType.getUuid())){
+                formMappings.get(formMapping).setVoided(true);
+            }
+        }
+        
```

Comment:
This logic is repeated multiple times in this commit. You should be moving this logic to a common service instead.